### PR TITLE
Add useTags config option support for java client code generation

### PR DESCRIPTION
**Pull request by pravussum**
Feature request: #10109

This PR adds a new configuration option "useTags" to the Java generator similar to the one that can be found e. g. in the Spring generator. 
The flag defaults to `true` to keep backwards compatibility (despite the default for the other generators is false). 
Setting it the `false` will prevent the generator to create one interface per tag (which might be not desirable). 

There has been a similar PR before, but it has been closed without comment: [10156](https://github.com/OpenAPITools/openapi-generator/pull/10156)

/cc Technical committee members @bbdouglas @sreeshas @jfiala @lukoyanov @cbornet @jeff9finger @karismann @Zomzog @lwlee2608

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
@pravussum thanks for the PR. Can you please resolve the merge conflicts when you've time and then PM me via Slack?

https://join.slack.com/t/openapi-generator/shared_invite/zt-12jxxd7p2-XUeQM~4pzsU9x~eGLQqX2g

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java [hunk 1]
```diff
@@ -214,9 +220,56 @@ public String getHelp() {
 
     @Override
     public void addOperationToGroup(String tag, String resourcePath, Operation operation, CodegenOperation co, Map<String, List<CodegenOperation>> operations) {
-        super.addOperationToGroup(tag, resourcePath, operation, co, operations);
-        if (MICROPROFILE.equals(getLibrary())) {
-            co.subresourceOperation = !co.path.isEmpty();
+        if (!useTags) {
+            String basePath = resourcePath;
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:224]: can you please add comment explaining what this code block does, ideally with an example?

**Comment by jekkel** [comment]
rebased @pravussum work in https://github.com/OpenAPITools/openapi-generator/pull/14615

**Comment by wing328** [comment]
closed via https://github.com/OpenAPITools/openapi-generator/pull/14794 instead

**Pull request closed without merging**